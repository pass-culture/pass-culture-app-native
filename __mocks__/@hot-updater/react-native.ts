import { ReactNode } from 'react'

const getUpdateSource = jest.fn(() => Promise.resolve({ source: 'source' }))

const HotUpdater = jest.fn(() => (app: ReactNode) => app)

export default Object.assign(HotUpdater, {
  getUpdateSource,
})
