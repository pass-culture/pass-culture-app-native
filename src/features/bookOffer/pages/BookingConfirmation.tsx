import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'
import styled from 'styled-components/native'

import { UseNavigationType } from 'features/navigation/RootNavigator'
import { _ } from 'libs/i18n'
import { ButtonPrimaryWhite } from 'ui/components/buttons/ButtonPrimaryWhite'
import { ButtonTertiaryWhite } from 'ui/components/buttons/ButtonTertiaryWhite'
import { GenericInfoPage } from 'ui/components/GenericInfoPage'
import { TicketBooked } from 'ui/svg/icons/TicketBooked'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'

export function BookingConfirmation() {
  const { goBack, navigate } = useNavigation<UseNavigationType>()

  return (
    <GenericInfoPage
      title={_(t`Réservation confirmée !`)}
      icon={TicketBooked}
      iconSize={getSpacing(65)}>
      <StyledBody>{_(t`Il te reste encore 140€ à dépenser sur le pass !`)}</StyledBody>
      <Spacer.Column numberOfSpaces={4} />
      <StyledBody>
        {_(
          t`Tu peux retrouver toutes les informations concernant ta réservation sur l’application`
        )}
      </StyledBody>
      <Spacer.Column numberOfSpaces={8} />
      <ButtonPrimaryWhite title={_(t`Voir ma réservation`)} onPress={goBack} />
      <Spacer.Column numberOfSpaces={4} />
      <ButtonTertiaryWhite
        title={_(t`Retourner à l'accueil`)}
        onPress={() => navigate('Home', { shouldDisplayLoginModal: false })}
      />
    </GenericInfoPage>
  )
}

const StyledBody = styled(Typo.Body).attrs({
  color: ColorsEnum.WHITE,
})({
  textAlign: 'center',
})
