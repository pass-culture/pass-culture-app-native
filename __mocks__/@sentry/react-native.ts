import * as ActualSentryModule from '@sentry/react-native'

const MockedSentryModule = jest.genMockFromModule<typeof ActualSentryModule>('@sentry/react-native')

module.exports = {
  ...MockedSentryModule,
}
