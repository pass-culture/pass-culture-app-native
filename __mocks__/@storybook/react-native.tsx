import React from 'react'; // @storybook
import { View } from 'react-native';

const configure = jest.fn();
const getStorybookUI = jest.fn().mockImplementation(() => () => <View />);

export { configure, getStorybookUI };
