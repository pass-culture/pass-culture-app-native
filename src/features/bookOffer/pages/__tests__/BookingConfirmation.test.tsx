import { render } from '@testing-library/react-native'
import React from 'react'

import { BookingConfirmation } from '../BookingConfirmation'

describe('<BookingConfirmation />', () => {
  it('should render correctly', () => {
    const page = render(<BookingConfirmation />)
    expect(page).toMatchSnapshot()
  })
})
