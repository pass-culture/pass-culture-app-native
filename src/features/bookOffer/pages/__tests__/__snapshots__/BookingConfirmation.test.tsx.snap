// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingConfirmation /> should render correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <View
      height="100%"
      width="100%"
    >
      undefined-SVG-Mock
    </View>
  </View>,
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "flexGrow": 1,
        "justifyContent": "center",
        "paddingHorizontal": 16,
      }
    }
  >
    <View>
      <View
        numberOfSpaces={18}
        style={
          Array [
            Object {
              "height": 72,
            },
          ]
        }
      />
      <View
        height={146}
        width={232}
      >
        undefined-SVG-Mock
      </View>
      <View
        numberOfSpaces={9}
        style={
          Array [
            Object {
              "height": 36,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-MediumItalic",
              "fontSize": 24,
              "lineHeight": 28,
              "textAlign": "center",
            },
          ]
        }
      >
        Réservation confirmée !
      </Text>
      <View
        numberOfSpaces={5}
        style={
          Array [
            Object {
              "height": 20,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Il te reste encore 140€ à dépenser sur le pass !
      </Text>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <Text
        color="#ffffff"
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 15,
              "lineHeight": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Tu peux retrouver toutes les informations concernant ta réservation sur l’application
      </Text>
      <View
        numberOfSpaces={8}
        style={
          Array [
            Object {
              "height": 32,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 24,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "maxWidth": 500,
            "opacity": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
            "width": "100%",
          }
        }
        testID="button-container"
      >
        <Text
          adjustsFontSizeToFit={false}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#eb0055",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 5,
                "maxWidth": 650,
              },
            ]
          }
          testID="button-title"
          textColor="#eb0055"
        >
          Voir ma réservation
        </Text>
      </View>
      <View
        numberOfSpaces={4}
        style={
          Array [
            Object {
              "height": 16,
            },
          ]
        }
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 24,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "maxWidth": 500,
            "opacity": 1,
            "paddingBottom": 2,
            "paddingLeft": 2,
            "paddingRight": 2,
            "paddingTop": 2,
            "width": "100%",
          }
        }
        testID="button-container"
      >
        <Text
          adjustsFontSizeToFit={false}
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 15,
                "lineHeight": 20,
                "marginLeft": 5,
                "maxWidth": 650,
              },
            ]
          }
          testID="button-title"
          textColor="#ffffff"
        >
          Retourner à l'accueil
        </Text>
      </View>
      <View
        customHeight={8}
        style={
          Array [
            Object {
              "height": 8,
            },
          ]
        }
      />
    </View>
  </RCTScrollView>,
]
`;
